lib/SPIRV/Mangler/Mangler.cpp

```diff
@@ -92,15 +92,15 @@ class MangleVisitor : public TypeVisitor {
 #else
     std::string MangledPrimitive =
         std::string(mangledPrimitiveString(T->getPrimitive()));
-    // out of all enums it makes sense to substitute only
-    // memory_scope/memory_order since only they appear several times in the
-    // builtin declaration.
-    if (MangledPrimitive == "12memory_scope" ||
-        MangledPrimitive == "12memory_order") {
+    // Builtin primitives such as int are not substitution candidates, but
+    // all other primitives are.  Even though most of these do not appear
+    // repeatedly in builtin function signatures, we need to track them in
+    // the substitution map.
+    if (T->getPrimitive() >= PRIMITIVE_STRUCT_FIRST) {
       if (!mangleSubstitution(T, mangledPrimitiveString(T->getPrimitive()))) {
         size_t Index = Stream.str().size();
         Stream << mangledPrimitiveString(T->getPrimitive());
-        Substitutions[Stream.str().substr(Index)] = SeqId++;
+        recordSubstitution(Stream.str().substr(Index), SeqId++);
```

Comment:
`recordSubstitution` is a pretty simple function which just replaces call to `operator[]` - would it make sense to move usages of `SeqId` into it too? As I can see it is not used anywhere else

Reply:
Good point, done!